apps/dashboard/src/components/workflow-editor/steps/email/email-editor.tsx

```diff
@@ -2,15 +2,17 @@ import { Separator } from '@/components/primitives/separator';
 import { getComponentByType } from '@/components/workflow-editor/steps/component-utils';
 import { EmailPreviewHeader } from '@/components/workflow-editor/steps/email/email-preview';
 import { EmailTabsSection } from '@/components/workflow-editor/steps/email/email-tabs-section';
-import { type UiSchema } from '@novu/shared';
+import { UiSchemaGroupEnum, type UiSchema } from '@novu/shared';
 
 const subjectKey = 'subject';
 const emailEditorKey = 'emailEditor';
 
-type EmailEditorProps = { uiSchema: UiSchema };
-export const EmailEditor = (props: EmailEditorProps) => {
-  const { uiSchema } = props;
-  const { [emailEditorKey]: emailEditor, [subjectKey]: subject } = uiSchema?.properties ?? {};
+export const EmailEditor = ({ uiSchema }: { uiSchema: UiSchema }) => {
+  if (uiSchema.group !== UiSchemaGroupEnum.EMAIL) {
```

Comment:
I suggest moving this logic outside the editor to the code that decides which group corresponds to each component. It's a bit of a code smell that the email editor needs to check for the enum.

It might not be necessary due to the existence of `getComponentByType`.

Reply:
Email editor is the component that decides which group corresponds to each component. The only other place where this could live is in `<EmailTabs/>` which is arguably worse place.